service/db.go

```diff
@@ -46,3 +54,62 @@ func (db *LevelDB) Delete(key []byte) error {
 func (db *LevelDB) Iterator() iterator.Iterator {
 	return db.DB.NewIterator(nil, db.ro)
 }
+
+type ProofsDatabase struct {
+	db     *leveldb.DB
```

Comment:
i think it's more consistent to reuse LevelDB and its interface here.

Reply:
I'd rather remove the LevelDB wrapper altogether as it only wraps `leveldb.DB` in order to not have to provide ro/wo...

Reply:
what's the problem with hiding ro/wo? 

anyway, i am fine with you removing LevelDB in this PR.
my main concern is that you are creating a new way to use level db in this repo, which make the code confusing.

Reply:
>what's the problem with hiding ro/wo?

Hiding ro/wo in the wrapper struct limits all read/write operations to use the same ro/wo. It might not always be optimal and necessary.

Additionally, it requires providing a wrapper method for every method of `leveldb.DB` that needs to be used.

IMO, the benefit of NOT having to explicitly set ro/wo on each operation is illusory and can actually be harmful (hurting performance in the best case or introducing bugs in the worst case - when an improper ro/wo is used).

I planned to remove the wrapper in #181 anyway but I can do it immediately.